src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -560,16 +560,26 @@ void MacroAssembler::call_VM_helper(Register oop_result, address entry_point, in
 
 // Maybe emit a call via a trampoline.  If the code cache is small
 // trampolines won't be emitted.
```

Comment:
> "The header comment is incorrect: "If the code cache is small trampolines won't be emitted."

I feel that Vladimir's comment is still legit here. From my reading, trampoline is avoidable because of  either codecache is small (far_branches() is false) or target_needs_far_branch() is false for runtime reloc.

Reply:
I've created [JDK-8286056](https://bugs.openjdk.java.net/browse/JDK-8286056) to clarify uses of `far_call`/`far_jump` and [JDK-8286058](https://bugs.openjdk.java.net/browse/JDK-8286058) to clarify uses of `trampoline_call`.